spec/lib/common/json-schema-validator-spec.js

```diff
@@ -53,44 +53,173 @@ describe('JsonSchemaValidator', function () {
+    describe('validate', function() {
+        it('should return true when validation pass', function () {
+            var result = validator.validate(testSchema1, { repo : '/172.31.128.1/mirrors' });
+            expect(result).to.be.true;
+        });
+
+        it('should throw validation error with incorrect data format', function () {
+            expect(function () {
+                validator.validate(testSchema1, { repo : 'abc' });
+            }).to.throw(/JSON schema validation failed/);
+        });
     });
 
-    it('should throw validation error with incorrect data format', function () {
-        expect(function () {
-            validator.validate(testSchema1, { repo : 'abc' });
-        }).to.throw(/JSON schema validation failed/);
-    });
-
-    it('should add correct schema', function () {
-        expect(validator.addSchema(testSchema1, 'test1')).to.be.empty;
-        expect(validator.addSchema(testRefSchema1)).to.be.empty;
-        expect(validator.addSchema(testRefSchema2)).to.be.empty;
-    });
+    describe('addSchema/getSchema', function() {
+        it('should add correct schema', function () {
+            expect(validator.addSchema(testSchema1, 'test1')).to.be.empty;
+            expect(validator.addSchema(testRefSchema1)).to.be.empty;
+            expect(validator.addSchema(testRefSchema2)).to.be.empty;
+        });
 
-    it('should throw error when add duplicated key', function () {
-        validator.addSchema(testSchema1, 'test1');
-        expect(function () {
+        it('should throw error when add duplicated key', function () {
             validator.addSchema(testSchema1, 'test1');
-        }).to.throw(/schema with key or id "test1" already exists/);
-    });
+            expect(function () {
+                validator.addSchema(testSchema1, 'test1');
+            }).to.throw(/schema with key or id "test1" already exists/);
+        });
 
-    it('should get existing schema', function () {
-        validator.addSchema(testSchema1, 'test1');
-        expect(validator.getSchema('test1')).to.deep.equal(testSchema1);
+        it('should get existing schema', function () {
+            validator.addSchema(testSchema1, 'test1');
+            expect(validator.getSchema('test1')).to.deep.equal(testSchema1);
+        });
+
+        it('should not find not existing schema', function () {
+            expect(validator.getSchema('test2')).to.be.undefined;
+        });
+
+        it('should throw error when reference schema not added', function () {
+            validator.addSchema(testRefSchema2);
+            expect(function () {
+                validator.getSchema('/refschema/r2');
+            }).to.throw(/can't resolve reference /);
+        });
     });
 
-    it('should not find not existing schema', function () {
-        expect(validator.getSchema('test2')).to.be.undefined;
+    describe('validatePatternsSkipped', function() {
+        var skipPatterns = [ /{{\s*foo.\w+\s*}}/, /<%\s*bar.\d+\s*%>/ ];
```

Comment:
If you do want to match brace `{` sth `}` here. Should use backslash to escape char. `\{\{`